### PR TITLE
Feature/foundation rework

### PR DESCRIPTION
**Pull request by mattiasnilsen**
Fixed admin page and some other small things

### CODE REVIEW

File: app/helpers/admin_helper.rb [hunk 1]
```diff
@@ -15,21 +15,10 @@ def active_booking
...
-			if latest_lock.state.eql?("locked")
-				@is_locked = true
-				#call to active booking to get @group (can this be done better?)
-				active_booking
-				unless current_user.in_group?(@group.to_s) || current_user.admin?
-					render :file => "app/views/lights/lock.html", :status => :unauthorized
-				end
-			end
+	def is_locked
+		if $locked_until == nil
```
Comment (Kalior) [inline app/helpers/admin_helper.rb]: how about using `$locked_until ||= Time.now`?

File: app/controllers/admin_controller.rb [hunk 2]
```diff
@@ -6,49 +6,22 @@ class AdminController < ApplicationController
...
-				lock_state.save
-
-				if params[:lock_type].eql?("locked")
-					flash[:lock_state_changed] = "Hue is " + @lock_type
-				else
-					flash[:lock_state_changed] = "Hue is unlocked"
-				end
-
-				redirect_to :action => 'index'
+		shouldLock = params[:lock] == "true"
```
Comment (Kalior) [inline app/controllers/admin_controller.rb]: I'm being picky here, but the rest of the application uses snake_case

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -16,6 +16,9 @@
 //= require turbolinks
 //= require_tree .
 
+var lights = lights || []
+var isPartyOn = isPartyOn || false
```
Comment (Kalior) [inline app/assets/javascripts/application.js]: I can't see what good these do. These values won't be used in admin view, and in the lights view they are set in index.html.erb

**Review by Kalior — COMMENTED** [review]
Added some comments about the code, otherwise, it looks good.

File: app/helpers/admin_helper.rb [hunk 4]
```diff
@@ -15,21 +15,8 @@ def active_booking
...
-				#call to active booking to get @group (can this be done better?)
-				active_booking
-				unless current_user.in_group?(@group.to_s) || current_user.admin?
-					render :file => "app/views/lights/lock.html", :status => :unauthorized
-				end
-			end
-		end
-		@is_locked
-	end
+	def is_locked
```
Comment (lindskogen) [inline app/helpers/admin_helper.rb]: Minor comment, conventions, methods that return boolean value in ruby should have a question mark at the end like so: `is_locked?`

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -21,6 +20,16 @@
         <div class="it-logo"></div>
       </div>
     </header>
+    <% flash.each do |name, msg| %>
+      <% if msg.is_a?(String) %>
```
Comment (lindskogen) [inline app/views/layouts/application.html.erb]: What was the problem here? Our previous solution didn't work?

File: app/controllers/admin_controller.rb [hunk 6]
```diff
@@ -6,49 +6,22 @@ class AdminController < ApplicationController
...
-					flash[:lock_state_changed] = "Hue is " + @lock_type
-				else
-					flash[:lock_state_changed] = "Hue is unlocked"
-				end
-
-				redirect_to :action => 'index'
+		should_lock = params[:lock] == "true"
+		if should_lock
+			if @group
+				$locked_until = @exp_date
```
Comment (lindskogen) [inline app/controllers/admin_controller.rb]: SO much cleaner 😄  Good job!

**Review by lindskogen — COMMENTED** [review]
Small comments :)

File: app/views/layouts/application.html.erb (hunk 5 above)
Comment (mattiasnilsen) [reply app/views/layouts/application.html.erb]: Previous solution? I used this because it worked with foundation with no problems

**Pull request merged**